lib/kafka/consumer.rb

```diff
@@ -280,6 +280,21 @@ def each_batch(min_bytes: 1, max_wait_time: 5, automatically_mark_as_processed:
       end
     end
 
+    # @param min_bytes [Integer] the minimum number of bytes to read before
+    #   returning messages from the server; if `max_wait_time` is reached, this
+    #   is ignored.
+    # @param max_wait_time [Integer, Float] the maximum duration of time to wait before
+    #   returning messages from the server, in seconds.
+    # @param topics_offsets [Hash] the offsets to fetch from. By default it will
+    #   fetch from where the offset manager defaults too.
+    # @return batch [Kafka::FetchedBatch] a message batch fetched from Kafka.
+    def poll(min_bytes: 1, max_wait_time: 5, offsets: {})
```

Comment:
I mean the [SyncGroup](https://kafka.apache.org/protocol#The_Messages_SyncGroup) API that synchronizes partition assignments – basically, one of the consumers in a group is chosen as Leader and is asked to assign partitions to the others. This assignment can include custom metadata that's currently left blank in ruby-kafka, so it would be possible to pass the offset targets in there. Of course only the Leader would get to actually send the resets, and we won't know ahead of time which consumer will be chosen.

Given the fact that consumer groups need to be relatively small anyway (more than 32 would be a stretch IMO) I think it's probably not a huge deal if each consumer separately seeks _every_ partition, as long as they follow the same procedure. This would also mean that we don't _need_ to coordinate, as a consumer will just use the local seek targets with the partitions assigned to it. The assignments may change at any time, so it needs access to all the offsets (or be able to coordinate, e.g. via SyncGroup).